augur/filter.py

```diff
@@ -407,24 +418,27 @@ def run(args):
 
         #If didnt find any categories specified, all seqs will be in 'unknown' - but don't sample this!
         if len(seq_names_by_group)==1 and ('unknown' in seq_names_by_group or ('unknown',) in seq_names_by_group):
-            print("WARNING: The specified group-by categories (%s) were not found."%args.group_by,
+            print("WARNING: The specified group-by categories (%s) were not found."%groups,
                   "No sequences-per-group sampling will be done.")
-            if any([x in args.group_by for x in ['year','month']]):
+            if any([x in groups for x in ['year','month']]):
                 print("Note that using 'year' or 'year month' requires a column called 'date'.")
             print("\n")
         else:
             # Check to see if some categories are missing to warn the user
             group_by = set(['date' if cat in ['year','month'] else cat
-                            for cat in args.group_by])
+                            for cat in groups])
             missing_cats = [cat for cat in group_by if cat not in meta_columns]
```

Comment:
I would add an additional test to the conditional here like this:

```python
missing_cats = [cat for cat in group_by if cat not in meta_columns and cat != "dummy"]
```

The dummy category will always be missing and this early check will allow you to omit the other checks for "dummy" in the lines that follow.